fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java

```diff
@@ -2362,10 +2362,17 @@ public ParseNode visitAdminSetPartitionVersion(StarRocksParser.AdminSetPartition
     @Override
     public ParseNode visitKillStatement(StarRocksParser.KillStatementContext context) {
         NodePosition pos = createPos(context);
-        long id = Long.parseLong(context.INTEGER_VALUE().getText());
+        long id = context.connId != null ? Long.parseLong(context.connId.getText()) : -1;
+        String queryId = context.queryId != null ? ((StringLiteral) visit(context.queryId)).getStringValue() : null;
         if (context.QUERY() != null) {
-            return new KillStmt(false, id, pos);
+            if (queryId != null) {
+                return new KillStmt(queryId, pos);
+            }
+            return new KillStmt(id, pos);
         } else {
+            if (context.connId == null) {
+                throw new ParsingException(String.format("connection id %s should be a positive integer", queryId));
```

Comment:
```suggestion
                throw new ParsingException(String.format("connection id %s should be a positive integer", context.connId.getText()));
```
??

Reply:
this happens when `kill connection 'xxx'`.
this error means user should use sql like `kill connection integer`